### ISSUE TITLE
DSC not obtaining a DAT through docker

### ISSUE THREAD

**Issue by aitorcelaya** [FraunhoferISST/DataspaceConnector#591]
I am trying to connect the Dataspace Connector (v5.1.2 used for the Testbed) to a locally installed DAPS. The issue comes when the Dataspace Connector is launched with docker, however it is able to correctly obtain a DAT when it is launched with maven.

The address for the DSC is https://localhost:8080
The address for the DAPS is http://localhost:4567

This is the DAPS being used:
![image](https://user-images.githubusercontent.com/82808687/133450781-2f46d54c-9851-435a-812d-817981738d1f.png)

This is the response obtained when launched with docker:

In the ERROR, it shows "Failed to connect to localhost/127.0.0.1:4567"

![image](https://user-images.githubusercontent.com/82808687/133444802-2ff62e24-e74c-4316-b957-d4b3f45bff2d.png)

This is the response obtained when launched with maven:

![image](https://user-images.githubusercontent.com/82808687/133450012-fab55ded-df52-49cd-9cd6-f73b6ce7f2ab.png)

The DSC was launched as specified in the documentation: https://international-data-spaces-association.github.io/DataspaceConnector/Deployment/Build
For the docker build, option 1 (build and run docker image) was used.

**Comment by ghost** [comment]
Is the DAPS also running in parallel as Docker Container in the same setup?

**Comment by ghost** [comment]
If no, then reaching a localhost address on the actual operating system by the container itself from within the Docker VM could be problematic.

If yes, the DAPS Docker container-name would probably have to be used instead of localhost.

**Comment by aitorcelaya** [comment]
At the moment, both containers are launched in parallel in the same VirtualBox. 

I have a new issue now with the Docker DSC. I was re-creating the issue yesterday to make sure it was a consistent issue. When I relaunched the VM, I now have a different issue. The Docker DSC is not picking up the configuration from application.properties.

This is from the application properties:
keyStore -> the certificate being used is TestbedCert.p12 (does include aki/ski extensions)
![image](https://user-images.githubusercontent.com/82808687/133581321-d95e8c7a-fd85-45f8-8ef9-393630fdaeb5.png)

But the Docker DSC shows the following in the logs:
It is trying to access conf/keystore-localhost.p12. The certificate mentioned above TestbedCert.p12 is located in the conf folder as it should.
![image](https://user-images.githubusercontent.com/82808687/133581538-4e27c55e-98e6-4ff8-a645-733454904c0c.png)

Thank you again!

**Comment by ghost** [comment]
"The Docker DSC is not picking up the configuration from application.properties" -> config.json

At the first start of the Dataspace Connector it writes the settings from the config.json into its database. If the Dataspace Connector has a connection to a persistent database (e.g. Postgre), it loads the configuration stored there from the second start on instead of reloading it from config.json. This allows changes to the configuration via the APIs of the Dataspace Connector to be persisted independently of a restart of the connector, but has the disadvantage that manual changes to the config.json are not adopted any more. Please either remove the docker-volume of the Dataspace Connector once or alternatively make the changes to the configuration via the APIs (or use H2).

**Comment by aitorcelaya** [comment]
Both issues are now solved. I am currently able to obtain the information from the DAPS by calling its container as suggested above. Thank you!

**Comment by ghost** [comment]
👍
